### PR TITLE
Update free-programming-books-subjects.md

### PR DESCRIPTION
**Pull request by YashSalunke12**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -628,6 +629,11 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [ZeroMQ Guide](http://zguide.zeromq.org/page%3Aall)
 
 
+### object oriented programming
+
+* [Object Oriented Programming](https://www.cl.cam.ac.uk/teaching/0910/OOProg/OOP.pdf) - Dr Robert Harle (PDF)
```
Comment (shrn01) [inline books/free-programming-books-subjects.md:634]: Is this resource free to be accessed by people who are not part of that university?

Comment (YashSalunke12) [reply books/free-programming-books-subjects.md]: @eshellman changes done

Comment (eshellman) [reply books/free-programming-books-subjects.md]: check your spelling in the index, but please attend to my suggestions on the category to put this in.

Comment (YashSalunke12) [reply books/free-programming-books-subjects.md]: @eshellman  All done

**Comment by eshellman** [comment]
sorry for delay. see #9109
Thanks, and welcome to free-programming-books!

**Pull request merged**